.github/actions/update-stars/main.js

```diff
@@ -0,0 +1,131 @@
+import core from '@actions/core';
+import { readFileSync, writeFileSync } from 'node:fs';
+import { fetch } from 'undici';
```

Comment:
actually, let's just set the node version to 20 in the CI. then `fetch` will be available as a global. https://nodejs.org/dist/latest-v20.x/docs/api/globals.html#fetch
```suggestion
```